draft-ietf-core-echo-request-tag.md

```diff
@@ -144,9 +144,7 @@ The server may use request freshness provided by the Echo option to verify the a
 
 Upon receiving a 4.01 Unauthorized response with the Echo option, the client SHOULD resend the original request with the addition of an Echo option with the received Echo option value. The client MAY send a different request compared to the original request. Upon receiving any other response with the Echo option, the client SHOULD echo the Echo option value in the next request to the server. The client MAY include the same Echo option value in several different requests to the server.
 
-A client MUST only send Echo values to endpoints it received them from (where as defined in {{RFC7252}} Section 1.2, the security association is part of the endpoint).
-In OSCORE processing, that means sending Echo values from outer options back in outer options,
-and those from inner options in inner options in the same security context.
+A client MUST only send Echo values to endpoints it received them from (where as defined in {{RFC7252}} Section 1.2, the security association is part of the endpoint). In OSCORE processing, that means sending Echo values from outer options back in outer options, and those from inner options in inner options in the same security context. Echo options in error responses not protected by OSCORE are treated as outer options.
```

Comment:
That might be read as giving license to a client to take the inner Echo from an inner error and use it on the outside, I don't think we want that. What motivates that addition anyway?

Reply:
I don't see how it could be read as a licence to a client to take the inner Echo from an inner error and use it on the outside. "not protected by OSCORE" means there are no inner.

The motivation was the following

```
C                                           P                                                      S

OSCORE Req    
----------------------------> 

error with echo (not OSCORE)
<---------------------------- 

OSCORE Req Outer Echo .           OSCORE Req without Echo
---------------------------->    ----------------------------> 
```

I do not think that RFC 8613 is clear the an echo in non-OSCORE error counts as Outer. And they "In OSCORE processing, that means sending Echo values from outer options back in outer options, and those from inner options in inner options in the same security context." Makes it unclear how to do when a Proxy responds with error to an OSCORE request.

Reply:
OK, then let's just say "from outer options (or non-OSCORE messages)"; merging with that on top of it.